### DISCUSSION TITLE
Is it possible to native HTML form submit?

### DISCUSSION THREAD

**Discussion by thkim9012** [react-hook-form/react-hook-form#8864] [Q&A] [answered]
I want to submit form to specific url (Legacy jsp server) with form data by post.

Now, createRoot and render React.createElement form, React.createElement input..
It is working but has some troubles that I don't know, 
so I want to make it using react-hook-form.

I don't need any validation, 
just create dynamic input name and value and action url..

**Comment by LauSam09** [comment]
If you don't need validation (or presumably other useful things like `dirty` state of fields) provided by `react-hook-form`, can't you just use a vanilla react form?

Minimal example:

```jsx
export default function App() {
  const handleSubmit = async (e) => {
    e.preventDefault();
    const formData = new FormData(e.currentTarget);
    await fetch("some-url", { method: "POST", body: formData });
  };

  return (
    <div className="App">
      <form onSubmit={handleSubmit}>
        <input name="field-one" />
        <button type="submit">Send</button>
      </form>
    </div>
  );
}
```

https://codesandbox.io/s/vanilla-react-form-zuv8xn

**Comment by LauSam09** [comment, marked as answer]
On the other hand, if you really do want to use RHF, you can do this:

```jsx
export default function App() {
  const { register, handleSubmit } = useForm();

  // The underlying form submission event is the 2nd argument supplied to `handleSubmit`:
  const onSubmit = async (_, e) => {
    e.preventDefault();
    const formData = new FormData(e?.target);
    await fetch("some-url", { method: "POST", body: formData });
  };

  return (
    <div className="App">
      <form onSubmit={handleSubmit(onSubmit)}>
        <input {...register("field-one")} />
        <button type="submit">Send</button>
      </form>
    </div>
  );
}
```

https://codesandbox.io/s/rhf-form-data-xno0w5

**Reply by thkim9012** [reply]
Thanks very much. it is very helpful.
